rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCostAnnotator.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import akka.actor.Actor
+import akka.actor.ActorLogging
+import akka.annotation.InternalApi
+import akka.cluster.Cluster
+import akka.cluster.ClusterEvent
+import akka.cluster.Member
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.ConnectionContext
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.HttpsConnectionContext
+import akka.pki.kubernetes.PemManagersProvider
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.security.KeyStore
+import java.security.SecureRandom
+import javax.net.ssl.KeyManager
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import scala.collection.immutable
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+@InternalApi
+final class PodDeletionCostAnnotator extends Actor with ActorLogging {
+  private val cluster = Cluster(context.system)
+  private val k8sSettings = KubernetesSettings(context.system)
+
+  private val http = Http()(context.system)
+  private lazy val apiToken = readConfigVarFromFilesystem(k8sSettings.apiTokenPath, "api-token").getOrElse("")
+  private val podNamespace = k8sSettings.podNamespace
+    .orElse(readConfigVarFromFilesystem(k8sSettings.podNamespacePath, "namespace"))
+    .getOrElse("default")
+
+  /**
+   * This uses blocking IO, and so should only be used to read configuration at startup.
+   */
+  private def readConfigVarFromFilesystem(path: String, name: String): Option[String] = {
```

Comment:
I think this should be passed in as a constructor parameter instead. The blocking IO isn't that nice. We can move this reading to the PodDeletionCost extension and run it from a Future with Akka's default `blocking-dispatcher`. When it has read the file it can start the actor. No need to start it immediately.

Reply:
using the `blocking-io-dispatcher` now as suggested. tks